lib/eu_central_bank.rb

```diff
@@ -136,6 +147,13 @@ def import_rates(format, s, opts = {})
     self
   end
 
+  def check_currency_available(currency)
+    currency_string = currency.to_s
+    return true if currency_string == "EUR"
+    return true if CURRENCIES.include?(currency_string)
+    raise CurrencyUnavailable.new(currency_string)
```

Comment:
Yes, that's better, but I don't think there's any need for additional behaviour (like `initialize` and `to_s` methods) apart from:

```ruby
raise CurrencyUnavailable, "No rates available for #{currency_string}"
```

You might wanna use that when you're returning more than just an error string, which is not the case here